src/core/shape/2d_primitives.js

```diff
@@ -618,17 +618,16 @@ p5.prototype.rect = function() {
  */
 p5.prototype.square = function(x, y, s, tl, tr, br, bl) {
   p5._validateParameters('square', arguments);
-  return this._renderRect.apply(this, arguments);
+  // duplicate width for height as only one value given
+  let args = Array.prototype.slice.call(arguments, 0, 3);
+  args.push(arguments[2]);
+  args = args.concat(Array.prototype.slice.call(arguments, 4));
+  return this._renderRect.apply(this, args);
 };
 
 // internal method to have renderer draw a rectangle
 p5.prototype._renderRect = function() {
   if (this._renderer._doStroke || this._renderer._doFill) {
-    // duplicate width for height if only one value given
-    if (arguments.length === 3) {
-      arguments[3] = arguments[2];
-    }
-
```

Comment:
This check is in place because `rect()` also takes a 3 argument overload which draws a square. It should be moved or something similar implemented in `rect()` itself.

Reply:
Oh, I thought to remove it because of the absence of example in the example section of `rect()` method with 3 arguments. Thought that it was done for overloading `square()`. 
So I guess after doing this, an example of rectangle with 3 arguments needed to be implemented in the docs.
What's your thought?

Reply:
@AnuragGupta93 I don't think an extra example for `rect()` with 3 arguments is necessary, the function signature should take care of it and an extra example is a bit much. But that's just me.

Reply:
Agreed